### PR TITLE
add snappy compression

### PR DESCRIPTION
**Pull request by ggriffiths**
for those who want to use snappy compression

### CODE REVIEW

**Comment by ggriffiths** [comment]
signed the CLA

File: rpc_util.go [hunk 1]
```diff
@@ -104,6 +105,26 @@ func (c *gzipCompressor) Type() string {
 	return "gzip"
 }
 
+// NewSnappyCompressor creates a Compressor based on Snappy.
+func NewSnappyCompressor() Compressor {
+	return &gzipCompressor{}
```
Comment (Thomasdezeeuw) [inline rpc_util.go]: Shouldn't this be `snappyCompressor`?

Comment (ggriffiths) [inline rpc_util.go]: oops, yes my bad

**Comment by iamqizhao** [comment]
There is no way gRPC-Go can support all kinds of compression algorithms. We only have plan to support a small number in standard Golang library (e.g., gzip, deflate). Snappy support can move to the contrib repo we will add shortly.

**Comment by zellyn** [comment]
Hmmmm. Given that GRPC started as a sort of stubby-alike, I'd think a zippy-alike compression fits in just fine. In fact, it should probably be part of the spec…

**Comment by iamqizhao** [comment]
One reason I do not want to put this into gRPC-Go repo because it introduces a new dependency on another github repo, which is the thing I have tried hard to minimize. I think it should go to contrib repo instead.

**Comment by ggriffiths** [comment]
Makes sense if that's the spec. Looking forward to adding it to the contrib repo.

**Pull request closed without merging**